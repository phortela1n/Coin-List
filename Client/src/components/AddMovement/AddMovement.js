import React from "react";
import Header from "../Common/Header/Header";
import SubMenu from "../Common/SubMenu/SubMenu";
import { Container } from "@material-ui/core/";
import LogOut from "../LogOut/LogOut";
import Profile from "../Profile/Profile";

function AddMovement() {
  return (
    <>
      <Header />
      <Container maxWidth="sm" className="big-container">
        <SubMenu />
        <Profile />
        <Container
          maxWidth="sm"
          className="big-container__coin-container"
        ></Container>
      </Container>
    </>
  );
}

export default AddMovement;
