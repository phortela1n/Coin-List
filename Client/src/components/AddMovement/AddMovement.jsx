import React from "react";
import TwoPhasesStepper from "./TwoPhasesStepper/TwoPhasesStepper";
import { useAuth0 } from "@auth0/auth0-react";
import Header from "../Common/Header/Header";
import SubMenu from "../Common/SubMenu/SubMenu";
import NoAuthorized from "../Common/NoAuthorized/NoAuthorized";
import { Container } from "@material-ui/core/";
import { CircularProgress } from "@material-ui/core";

export default function AddMovement() {
  const { isAuthenticated, isLoading } = useAuth0();

  /**
   * FORM STATE
   **/

  // Radio State
  const [OperationType, setOperationType] = React.useState("Buy");
  const handleChange = (event) => {
    setOperationType(event.target.value);
  };

  // Callendar State
  const [selectedDate, setSelectedDate] = React.useState(new Date());

  const handleDateChange = (date) => {
    setSelectedDate(date);
  };

  //Inputs State

  //QUANTITY

  const [quantityValues, setquantityValues] = React.useState({
    quantity: "",
  });

  const handleQuantity = (prop) => (event) => {
    setquantityValues({ ...quantityValues, [prop]: event.target.value });
  };

  //PRICE

  const [priceValues, setpriceValues] = React.useState({
    amount: "",
  });

  const handlePrice = (prop) => (event) => {
    setpriceValues({ ...priceValues, [prop]: event.target.value });
  };

  /**
   * END FORM STATE
   **/

  /*
   **Handle Click confirm form
   */
  function handleMovementsClick() {
    console.log(selectedDate);
    let newMovement = {
      type: OperationType,
      buyPrice: priceValues.amount,
      quantity: quantityValues.quantity,
      date: selectedDate.toISOString().substring(0, 10),
    };
    console.log(newMovement);
  }

  if (isLoading) {
    return (
      <center>
        <>
          <CircularProgress className="loadcircle" color="secondary" />
        </>
      </center>
    );
  }
  return (
    (isAuthenticated && (
      <>
        <Header />
        <Container
          maxWidth="sm"
          className="big-container big-container--addCoin"
        >
          <SubMenu changeValue={2} />
          <Container
            maxWidth="sm"
            className="big-container__coin-container big-container__movements"
          >
            <TwoPhasesStepper
              selectedDate={selectedDate.toISOString().substring(0, 10)}
              handleDateChange={handleDateChange}
              value={OperationType}
              handleChange={handleChange}
              quantityValues={quantityValues}
              handleQuantity={handleQuantity}
              priceValues={priceValues}
              handlePrice={handlePrice}
              handleMovementsClick={handleMovementsClick}
            />
          </Container>
        </Container>
      </>
    )) || <NoAuthorized />
  );
}
