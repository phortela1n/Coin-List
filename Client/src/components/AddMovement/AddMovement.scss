.big-container--addMovement {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.addMovement-box {
  display: flex;
  flex-direction: column;
}

.button-box {
  display: flex;
  justify-content: center;
  padding-top: 30px;
}

.AddMovement-box--reset-status {
  height: 160px !important;
  padding-top: 10px;
}
