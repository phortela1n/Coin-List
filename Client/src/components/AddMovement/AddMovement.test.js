import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import { AddMovement } from './AddMovement';
import { useAuth0 } from "@auth0/auth0-react";
import { BrowserRouter as Router } from "react-router-dom";


//  para que en todos los tests devuelva la misma respuesta
// jest.mock("@auth0/auth0-react", () => ({
//   useAuth0: () => ({user : null, isAuthenticated: null, isLoading: null})
// }));


jest.mock('@auth0/auth0-react');
jest.spyOn(console, 'error').mockImplementation(() => { });


test('Should redirect to login when user is not Authenticated', () => {
  useAuth0.mockReturnValue({ user: null, isAuthenticated: false, isLoading: false})
  const { asFragment } = render(<AddMovement userCoins={[]} />)
  expect(asFragment()).toMatchSnapshot()
  expect(screen.getByText('Log In')).toBeInTheDocument(); // existencia
  expect(screen.queryByText('Movement details')).toBeNull() // no existencia
});

test('AddMovement renders correctly', () => {
  useAuth0.mockReturnValue({ user: {email: 'foo@bar.com' }, isAuthenticated: true, isLoading: false})
  const { asFragment} = render(
    <Router>
      <AddMovement userCoins={[]} dispatch={() => { }} />
    </Router>
  )
  expect(asFragment()).toMatchSnapshot()
  expect(screen.getByText('Movement details')).toBeInTheDocument()
});

test('Should handle date selection', () => {
   useAuth0.mockReturnValue({ user: {email: 'foo@bar.com' }, isAuthenticated: true, isLoading: false})
  const selectedDate = new Date(2019, 8, 15)
  const { asFragment} = render(
    <Router>
      <AddMovement userCoins={[]} dispatch={() => { }} selectedDate={selectedDate} />
    </Router>
  )
  expect(asFragment()).toMatchSnapshot()
});



  // console.error.mockRestore();