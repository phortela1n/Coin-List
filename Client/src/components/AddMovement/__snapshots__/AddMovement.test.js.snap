// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should redirect to login when user is not Authenticated 1`] = `
<DocumentFragment>
  <section
    class="full-page"
  >
    <div />
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary test MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
          >
            CoinList
          </h6>
        </div>
      </header>
    </div>
    <div
      class="Container"
    >
      <video
        autoplay=""
        class="Video"
        loop=""
      >
        <source
          src="https://assets.mixkit.co/videos/preview/mixkit-rotating-shot-of-cryptocurrency-coins-over-dollar-bills-30660-large.mp4"
          type="video/mp4"
        />
        Your browser does not support the video tag.
      </video>
      <div
        class="Content"
      >
        <div
          class="SubContent"
        >
          <buttom>
            <button
              class="login-anchor pointer"
            >
              Log In
            </button>
          </buttom>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
